### PR TITLE
Test cases for cpuinfo and numa_cpu_map

### PR DESCRIPTION
**Pull request by Shrutikabansal**

### CODE REVIEW

File: netcontrold/tests/lib/test_util.py [hunk 1]
```diff
@@ -18,71 +18,170 @@
...
+physical id     : 0
+
+processor       : 3
+core id         : 1
+physical id     : 0
+"""
+
+_BASIC_CPU_INFO_Processorid_NULL = """
+processor       : 
+physical id     : 0
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:52]: Same as _BASIC_CPU_INFO_0_processor ? ..

File: netcontrold/tests/lib/test_util.py [hunk 2]
```diff
@@ -18,71 +18,170 @@
...
 
-_BASIC_CPU_INFO_1 = """
-processor		: 0
-core id 		: 0
-physical id 	: 0
 
-processor		: 1
-core id 		: 1
-physical id 	: 0
+_BASIC_CPU_INFO_coreid_NULL = """
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:59]: same as _BASIC_CPU_INFO_0_coreid ? The reason I quote the other one is that, we check for missing value there as well. So, one missing value is enough to raise the exception.

File: netcontrold/tests/lib/test_util.py [hunk 3]
```diff
@@ -18,71 +18,170 @@
...
 
-processor		: 3
-core id 		: 1
-physical id 	: 0
+_BASIC_CPU_INFO_physical_NULL = """
+processor       : 
+physical id     : 0
+core id         : 1
+processor       : 
+physical id     : 0
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:74]: same as in _BASIC_CPU_INFO_0_physicalid ?

File: netcontrold/tests/lib/test_util.py [hunk 4]
```diff
@@ -18,71 +18,170 @@
...
-            mock.mock_open(read_data=_BASIC_CPU_INFO_1).return_value
+            mock.mock_open(read_data=_BASIC_CPU_INFO_Processorid_NULL).return_value
+        ]
+        self.assertRaises(ValueError, util.cpuinfo)
+       
+
+    @mock.patch('netcontrold.lib.util.open')
+    def test_numa_cpu_map_coreid(self, mock_open):
+        mock_open.side_effect = [
+            mock.mock_open(read_data=_BASIC_CPU_INFO_Processorid_NULL).return_value
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py]: I guess you mean mock.mock_open(read_data=_BASIC_CPU_INFO_Coreid_NULL).return_value

File: netcontrold/tests/lib/test_util.py [hunk 5]
```diff
@@ -18,71 +18,170 @@
...
+        ]
+        self.assertRaises(ValueError, util.cpuinfo)
+
+
+    @mock.patch('netcontrold.lib.util.open')
+    def test_numa_cpu_map_physical(self, mock_open):
+        mock_open.side_effect = [
+            mock.mock_open(read_data=_BASIC_CPU_INFO_Processorid_NULL).return_value
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py]: And physicalid check here..

**Review by gmuthukr — CHANGES_REQUESTED** [review]
Overall, I ack positive and negative testcases. For other field specific tests, I wish you rebase your master and rework on changes to avoid duplicated tests. So, send positive and negative tests in one PR for now, after rebase.

File: netcontrold/tests/lib/test_util.py (hunk 1 above)
Comment (Shrutikabansal) [reply netcontrold/tests/lib/test_util.py:52]: Yes, it is duplicate but i am not getting updated test cases in my branch

**Pull request closed without merging**